lxd/container_lxc.go

```diff
@@ -332,7 +332,7 @@ func containerLXCCreate(s *state.State, args db.InstanceArgs) (instance.Instance
 	}
 
 	// Create a new database entry for the container's storage volume
-	_, err = s.Cluster.StoragePoolVolumeCreate(args.Project, args.Name, "", storagePoolVolumeTypeContainer, false, poolID, volumeConfig)
+	_, err = s.Cluster.StoragePoolVolumeCreate(args.Project, args.Name, "", storagePoolVolumeTypeContainer, args.Snapshot, poolID, volumeConfig)
```

Comment:
@monstermunchkin good spotting. Can you hypothesise as to why this hasn't been an issue before?

Also, just a nit, but could you use `c.IsSnapshot()` here, as further down in this function there is more snapshot specific logic that uses `c.IsSnapshot()` and this would keep it consistent (in case the use of args.Snapshot changes in the future for some reason).

Reply:
@monstermunchkin you mentioned in your commit msg about this bug causing migration failures, could you elaborate for my own education? :)

Reply:
@tomponline Apparently, this doesn't cause an issue anymore. We should still keep the commit (but use `c.IsSnapshot()`), and I'll fix the commit message.